src/resources/elements/formAddressInput/formAddressInput.ts

```diff
@@ -47,9 +47,7 @@ export class FormAddressInput {
           } else {
             const address = await this.ensService.getAddressForEns(newValue);
             if (address) {
-              this.ens = newValue;
-              this.ignoreNewValue = true;
-              this.value = address;
+              this.ens = address;
```

Comment:
on initial address input, both `getEnsForAddress` and  `getAddressForEns` would fire, in that order. the latter would cache its ` getAddressForEns(ens: string)` `ens` argument, but which in this case was the address rather than ENS name. this resulted in one of the caches looking like so:
```ens: address,
address:address
``` instead of
```ens:address,
```address:ens
``` and the functional result was the following sequences of actions/result
1. enter address, display ens result
2. enter ens, display address result
3.enter address, display address result

making each function cache it's respective bit seemed to have helped